reverse_file.py

```diff
@@ -1 +1,53 @@
+import os
 
+
+def file_length(name: str) -> int:
+    length = os.stat(name).st_size
+    return length
+
+
+def read_file(name: str, offset: int, size: int, output_buf_offset: int) -> bytearray:
```

Comment:
Although returning a bytearray seems more pythonic and it is better API in general, the original API (where `output_buf` is the argument) allows to avoid additional memory allocations and read data directly into the provided buffer

Reply:
Is it supposed to work like that?
`def read_file(name: str, offset: int, size: int, output_buf: bytearray, output_buf_offset: int) -> None:
    """
    Reads `size` bytes from file `name` starting from `offset` index and
    stores them into `output_buf` starting from `output_buf_offset`
  """
    with open(name, "rb") as f:
        f.seek(offset)
        chunk = f.read(size)
        output_buf[:output_buf_offset] = chunk`

Reply:
```def read_file(name: str, offset: int, size: int, output_buf: bytearray, output_buf_offset: int) -> None:
    """
    Reads `size` bytes from file `name` starting from `offset` index and
    stores them into `output_buf` starting from `output_buf_offset`
  """
    with open(name, "rb") as f:
        f.seek(offset)
        chunk = f.read(size)
        output_buf[:output_buf_offset] = chunk```

Reply:
Is it supposed to work like that?

Reply:
```def read_file(name: str, offset: int, size: int, output_buf: bytearray, output_buf_offset: int) -> None:
    """
    Reads `size` bytes from file `name` starting from `offset` index and
    stores them into `output_buf` starting from `output_buf_offset`
  """
    with open(name, "rb") as f:
        f.seek(offset)
        chunk = f.read(size)
        output_buf[:output_buf_offset] = chunk```